### ISSUE TITLE
Segmentation fault for some invalid PNG input

### ISSUE THREAD

**Issue by Zrzzzz** [pts/imgdataopt#4]
# Environment
Distributor ID: Ubuntu
Description: Ubuntu 18.04.6 LTS
Release: 18.04
Codename: bionic

# Review Command

`./imgdataopt crash 123.pnm`

# Attach

https://drive.google.com/file/d/1D2ThZgA3Vb0RUUPW--K2jcJ_mBOiu67B/view?usp=share_link

# Description

Here is `valgrind`'s output

```
==2875== Memcheck, a memory error detector
==2875== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==2875== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==2875== Command: ../imgdataopt collect/out:id:000000,sig:11,src:000082,op:ext_AO,pos:35 123.pnm
==2875== 
==2875== Invalid write of size 8
==2875==    at 0x10DAD7: ??? (in imgdataopt/fuzzing/sys/imgdataopt)
==2875==    by 0x109F13: ??? (in imgdataopt/fuzzing/sys/imgdataopt)
==2875==    by 0x4E5FC86: (below main) (libc-start.c:310)
==2875==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==2875== 
==2875== 
==2875== Process terminating with default action of signal 11 (SIGSEGV)
==2875==  Access not within mapped region at address 0x0
==2875==    at 0x10DAD7: ??? (in imgdataopt/fuzzing/sys/imgdataopt)
==2875==    by 0x109F13: ??? (in imgdataopt/fuzzing/sys/imgdataopt)
==2875==    by 0x4E5FC86: (below main) (libc-start.c:310)
==2875==  If you believe this happened as a result of a stack
==2875==  overflow in your program's main thread (unlikely but
==2875==  possible), you can try to increase the size of the
==2875==  main thread stack using the --main-stacksize= flag.
==2875==  The main thread stack size used in this run was 8388608.
==2875== 
==2875== HEAP SUMMARY:
==2875==     in use at exit: 357,258 bytes in 2 blocks
==2875==   total heap usage: 3 allocs, 1 frees, 361,354 bytes allocated
==2875== 
==2875== 552 bytes in 1 blocks are still reachable in loss record 1 of 2
==2875==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==2875==    by 0x4EBCDF9: __fopen_internal (iofopen.c:65)
==2875==    by 0x4EBCDF9: fopen@@GLIBC_2.2.5 (iofopen.c:89)
==2875==    by 0x109D56: ??? (in imgdataopt/fuzzing/sys/imgdataopt)
==2875==    by 0x4E5FC86: (below main) (libc-start.c:310)
==2875== 
==2875== 356,706 bytes in 1 blocks are still reachable in loss record 2 of 2
==2875==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==2875==    by 0x10A7BD: ??? (in imgdataopt/fuzzing/sys/imgdataopt)
==2875==    by 0x10A8F0: ??? (in imgdataopt/fuzzing/sys/imgdataopt)
==2875==    by 0x10D966: ??? (in imgdataopt/fuzzing/sys/imgdataopt)
==2875==    by 0x109F13: ??? (in imgdataopt/fuzzing/sys/imgdataopt)
==2875==    by 0x4E5FC86: (below main) (libc-start.c:310)
==2875== 
==2875== LEAK SUMMARY:
==2875==    definitely lost: 0 bytes in 0 blocks
==2875==    indirectly lost: 0 bytes in 0 blocks
==2875==      possibly lost: 0 bytes in 0 blocks
==2875==    still reachable: 357,258 bytes in 2 blocks
==2875==         suppressed: 0 bytes in 0 blocks
==2875== 
==2875== For counts of detected and suppressed errors, rerun with: -v
==2875== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
[1]    2875 segmentation fault  valgrind --leak-check=full --show-leak-kinds=all ../imgdataopt  123.pnm
```

**Comment by pts** [comment]
Thank you for reporting this bug! imgdataopt (`./imgdataopt issue4.png issue4.out.pnm`) indeed crashes for that input with Segmentation fault. Please note that the input file is not valid PNG, because it has a CRC error.

[issue4.png](https://user-images.githubusercontent.com/379746/225110387-21b31cd2-1427-4f29-bc38-6264226b2aa1.png)

**Comment by pts** [comment]
Fixed in 413269e4ecc87be0ecefd89a426c624b3101efcb.
